odd-collector/odd_collector/adapters/postgresql/mappers/relationships/cardinality_checker.py

```diff
@@ -0,0 +1,49 @@
+from functools import cached_property
+
+from itertools import chain
+from odd_models.models.models import CardinalityType, DataSetField
+
+from adapters.postgresql.models import UniqueConstraint
+
+
+class CardinalityChecker:
+    def __init__(self,
+                 ref_fk_field_list: list[DataSetField],
+                 unique_constraints: list[UniqueConstraint]):
+        """
+        :param ref_fk_field_list: list of DataSetField of referenced foreign key
+        :param unique_constraints: list of UniqueConstraint of target table
+        """
+        self.ref_fk_field_list = ref_fk_field_list
+        self.unique_constraints = unique_constraints
+
+    def get_cardinality(self) -> CardinalityType:
+        if self._is_ref_to_unique and self._is_ref_to_nullable:
+            return CardinalityType.ONE_TO_ZERO_OR_ONE
+
+        if self._is_ref_to_unique and (not self._is_ref_to_nullable):
+            return CardinalityType.ONE_TO_EXACTLY_ONE
+
+        if self._is_ref_to_nullable and (not self._is_ref_to_unique):
```

Comment:
@AndreyNenashev don't you think that it is clearer not to change the order of logical elements? So with remaining order it would be: `if (not self._is_ref_to_unique) and self._is_ref_to_nullable:`